dev/tests/integration/testsuite/Magento/Catalog/Model/ProductRepositoryTest.php

```diff
@@ -233,4 +233,24 @@ public function testCustomLayout(): void
         }
         $this->assertTrue($caughtException);
     }
+
+    /**
+     * Tests product repository update should use provided store code.
+     *
+     * @magentoDataFixture Magento/Catalog/_files/product_simple.php
+     */
+    public function testProductUpdate(): void
+    {
+        $sku = 'simple';
+        $nameUpdated = 'updated';
+
+        $product = $this->productRepository->get($sku, false, 0);
+        $product->setName($nameUpdated);
+        $this->productRepository->save($product);
+        $product = $this->productRepository->get($sku, false, 0);
```

Comment:
Maybe provide some more assertions - I'm not sure about your goal but good example could be adding extra scope like local scope code and check if accidentaly the product name was not changed?

Reply:
done